src/gammar.rs

```diff
@@ -15,8 +17,66 @@ pub struct ErrorInfo {
     )>,
 }
 
-pub fn checkerror(local_path: &Path, source: &str, input: tree_sitter::Node) -> Option<ErrorInfo> {
-    checkerror_inner(local_path, &source.lines().collect(), input)
+pub async fn checkerror<'a>(
+    local_path: &Path,
+    source: &str,
+    input: tree_sitter::Node<'a>,
+    whole: bool,
+) -> Option<ErrorInfo> {
+    let future_cmake_lint = if whole {
+        run_cmake_lint(local_path).await
+    } else {
+        None
+    };
+
+    let mut result = checkerror_inner(local_path, &source.lines().collect(), input);
+    if let Some(v) = future_cmake_lint {
+        let error_info = result.get_or_insert(ErrorInfo { inner: vec![] });
+        for item in v.inner {
+            error_info.inner.push(item);
+        }
+    };
+
+    result
+}
+
+async fn run_cmake_lint(path: &Path) -> Option<ErrorInfo> {
```

Comment:
But.. seems this function does not have await
. Emmm

Reply:
I made this function async because I want to run external linter in parallel with the built-in one..